src/Transports/MassTransit.SqlTransport.PostgreSql/Configuration/PostgresSqlTransportConfigurationExtensions.cs

```diff
@@ -7,15 +7,16 @@ namespace MassTransit
 
     public static class PostgresSqlTransportConfigurationExtensions
     {
-        public static IServiceCollection AddPostgresMigrationHostedService(this IServiceCollection services, bool create = true, bool delete = false)
+        public static IServiceCollection AddPostgresMigrationHostedService(this IServiceCollection services, bool createDatabase = true, bool delete = false, bool createInfrastructure = true)
```

Comment:
I would prefer to change this to a callback overload rather that more booleans, just give them the Options in an action callback and let them set what they want. Pre-initialize the options to the default values, then let them customize those options in the callback.

Reply:
To be clear, leave the previously existing method alone, just add an additional one that in the Configure() for the options invokes the callback.